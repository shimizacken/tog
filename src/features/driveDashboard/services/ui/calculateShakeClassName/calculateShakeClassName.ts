export type ShakeStyles = {
  shakeveryveryfast?: string;
  shakeveryfast?: string;
  shakefast?: string;
  shakeslow?: string;
  shakeveryslow?: string;
};

export const calculateShakeClassName = (
  drivingStatus: boolean,
  throttleSpeed: number,
  styles: ShakeStyles
) => {
  if (drivingStatus === false) {
    return;
  }

  if (throttleSpeed > 80 && throttleSpeed <= 100) {
    return styles.shakeveryveryfast;
  }

  if (throttleSpeed > 60 && throttleSpeed <= 80) {
    return styles.shakeveryfast;
  }

  if (throttleSpeed > 40 && throttleSpeed <= 60) {
    return styles.shakefast;
  }

  if (throttleSpeed > 20 && throttleSpeed <= 40) {
    return styles.shakeslow;
  }

  if (throttleSpeed > 0 && throttleSpeed <= 20) {
    return styles.shakeveryslow;
  }

  return;
};
