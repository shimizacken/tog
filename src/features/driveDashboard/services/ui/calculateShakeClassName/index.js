const calculateShakeClassName = (drivingStatus, throttleSpeed, styles) => {

    if(drivingStatus === false) {
        return;
    }

    if(throttleSpeed > 80 && throttleSpeed <= 100) {
        return styles.shakeveryveryfast;
    }

    if(throttleSpeed > 60 && throttleSpeed <= 80) {
        return styles.shakeveryfast;
    }
    
    if(throttleSpeed > 40 && throttleSpeed <= 60) {
        return styles.shakefast;
    }

    if(throttleSpeed > 20 && throttleSpeed <= 40) {
        return styles.shakeslow;
    }

    if(throttleSpeed > 0 && throttleSpeed <= 20) {
        return styles.shakeveryslow;
    }

    return;
};

export default calculateShakeClassName;