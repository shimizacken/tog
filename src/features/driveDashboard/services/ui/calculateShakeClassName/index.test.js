import calculateShakeClassName from './index';

describe('calculateShakeClassName test suite', () => {
  
    it('should return undefined when no drivingStatus passed', () => {
        
        expect(calculateShakeClassName()).toMatchSnapshot();
    });
});
