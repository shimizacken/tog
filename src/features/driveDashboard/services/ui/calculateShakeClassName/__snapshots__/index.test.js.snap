// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateShakeClassName test suite should return undefined when no drivingStatus passed 1`] = `undefined`;
